pkg/dmaasbackup/nonperiodic_backup.go

```diff
@@ -121,44 +122,89 @@ func (d *dmaasBackup) cleanupNonPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) err
 		return nil
 	}
 
-	backupList, err := d.backupLister.List(
-		labels.SelectorFromSet(map[string]string{
-			v1alpha1.DMaaSBackupLabelKey:  dbkp.Name,
-			velerov1api.ScheduleNameLabel: schedule.ScheduleName,
-		}),
-	)
-
+	backupList, err := d.listScheduledBackups(dbkp, schedule.ScheduleName)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get list of backup for schedule=%s", schedule.ScheduleName)
 	}
 
+	if len(backupList) < retainCount {
+		d.logger.Debugf("Number of backups are %v, required %v backups to trigger cleanup",
+			len(backupList),
+			retainCount)
+		return nil
+	}
+
 	defer d.logger.Debug("Completed cleanup for non-periodic schedule")
 
 	sort.Sort(sort.Reverse(byBackupCreationTimeStamp(backupList)))
 
-	// We need to retain `fullBackupRetentionThreshold` number of backups
-	// check for completed backup count
-	var completedBackup int
+	requiredCompletedBackup := retainCount
+	completedIdx, successfulIdx := getBackupCleanupIdx(backupList, &requiredCompletedBackup)
```

Comment:
thinking of a variable name change for completedIdx and successfulIdx.. will think of and suggest the name

Reply:
ok.